### PR TITLE
German Go - btracker App API

### PR DESCRIPTION
**Pull request by germancin**
refactoring...

### CODE REVIEW

File: Answers.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+### Describe the following: DataBase, Collection , Document
+    Database: In Mongo a database is where we save the collections.
+    Collection: In Mongo a colection is where we store our documents.
+    Document: Basically is a schema that represents the information format in our collention.
+    
+
+### Describe how one can achieve the pattern of relationships in MongoDB. What needs to take place at the schema level? How do we 'fill' in the appropriate relational data using mongoose?
+    We need to determine which kind of relationship we want and after that we have to add the Objectid of the referenced
+    collection into the target collection.
```
Comment (dys2) [inline Answers.md]: To answer the "fill" in question, you would use populate along with the query.

File: Answers.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+### Explain a way to break up an API into Sub-Applications, which tool did we use to do that?
+    The way is separating in:
+    Controllers: Here we are have all our logic.
+    Models: Here we have the schema of the database and the connections with the Controllers.
+    Router: The router is where the endpoint where the front hit to it.
+    www: Here is where we have our front-end and this should hit our Router endpoints.
```
Comment (dys2) [inline Answers.md]: Nice!

File: api/models/categoryModel.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        required: [true, 'Please provide title.']
+    },
+    createdOn: {
+        type: String,
+        required: true,
+        default: new Date(),
+    }
+});
+
+module.exports = mongoose.model('categories', CategorySchema);
```
Comment (dys2) [inline api/models/categoryModel.js]: Typically for the collection name the nomenclature I'm familiar with is using a Capital letter and singular 'Category'.

File: api/controllers/budgetController.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const budgetModel = require('../models/budgetModel'); //mongoose instance
+
+const getBudget = (req, res) => {
+    budgetModel.find({})
+        .populate()
+        .exec((err, post) => {
+            console.log('hit getBudget');
```
Comment (dys2) [inline api/controllers/budgetController.js]: Reminder to remove console logs when finished with them

File: api/controllers/categoryController.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+const getCategories = (req, res) => {
+    categoryModel.find({})
+        .select('title')
+        .exec((err, resp) => {
+            res.status(200).send(resp)
+        });
+};
+
+const insertCategory = (req, res) => {
+    categoryModel.create(req.body)
```
Comment (dys2) [inline api/controllers/categoryController.js]: very trusting of what's in the req.body but on the flip side it's really smooth like that

File: api/routes/expenseRoutes.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+const express = require('express');
+const expenseController = require('../controllers/expenseController');
+let expenseRouter = express.Router();
+
+// EndPoint: /api/expense/
+
+expenseRouter.route('/').get(expenseController.getExpenses);
+
+expenseRouter.route('/').post(expenseController.insertExpense);
```
Comment (dys2) [inline api/routes/expenseRoutes.js]: you can actually chain on routes using the same path like
```
expenseRouter.route('/')
  .get(expenseController.getExpenses)
  .post(expenseController.insertExpense);
```

File: api/controllers/budgetController.js (hunk 4 above)
Comment (germancin) [reply api/controllers/budgetController.js]: done! 👍

File: api/routes/expenseRoutes.js (hunk 6 above)
Comment (germancin) [reply api/routes/expenseRoutes.js]: I'll add it to the code.

File: Answers.md (hunk 1 above)
Comment (germancin) [reply Answers.md]: I rephrased in a more eloquent way  I hope 👍

File: Answers.md (hunk 2 above)
Comment (germancin) [reply Answers.md]: thanks

**Pull request closed without merging**